nilearn/__init__.py

```diff
@@ -8,67 +8,18 @@
 
 See http://nilearn.github.com for complete documentation.
 """
+import gzip
 
-from version import __version__
+from .version import check_module_dependencies
 
+#
+check_module_dependencies()
+del check_module_dependencies
 
-def _import_module_with_version_check(module_name, minimum_version):
-    """Check that module is installed with a recent enough version
-    """
-    from distutils.version import LooseVersion
-
-    try:
-        module = __import__(module_name)
-    except ImportError as exc:
-        user_friendly_info = ('{} could not be found, '
-                              'please install it properly to use nilearn'
-                              ).format(module_name)
-        exc.args += (user_friendly_info,)
-        raise
-
-    module_version = module.__version__
-
-    version_too_old = (not LooseVersion(module_version) >=
-                       LooseVersion(minimum_version))
-
-    if version_too_old:
-        message = (
-            'A {module_name} version of at least {minimum_version} '
-            'is required to use nilearn. {module_version} was found. '
-            'Please upgrade {module_name}').format(
-                module_name=module_name,
-                minimum_version=minimum_version,
-                module_version=module_version)
-
-        raise ImportError(message)
-
-    return module
-
-
-def _check_dependencies():
-    _required_module_versions = [('numpy', '1.6.0'),
-                                 ('scipy', '0.9.0'),
-                                 ('sklearn', '0.10'),
-                                 ('nibabel', '1.1.0')]
-
-    for module_name, minimum_version in _required_module_versions:
-        _import_module_with_version_check(module_name, minimum_version)
-
-    try:
-        import gzip
-        if hasattr(gzip.GzipFile, 'max_read_chunk'):
-            # Monkey-patch gzip to have faster reads on large
-            # gzip files
-            gzip.GzipFile.max_read_chunk = 100 * 1024 * 1024  # 100Mb
-    except ImportError as exc:
-        exc.args += ('Python has been compiled without gzip, '
-                     'reading nii.gz files will be impossible.',)
-        raise
-
-_check_dependencies()
-
-del _import_module_with_version_check
-del _check_dependencies
+# Monkey-patch gzip to have faster reads on large
+# gzip files
+if hasattr(gzip.GzipFile, 'max_read_chunk'):
+    gzip.GzipFile.max_read_chunk = 100 * 1024 * 1024  # 100Mb
```

Comment:
... but making it mandatory is defenitely a good idea. Aprat from the .nii.gz format, a lot of datasets use this format.